### PR TITLE
Fix a collection of small issues

### PR DESCRIPTION
**Pull request by arcondello**

### CODE REVIEW

File: dimod/decorators.py [hunk 1]
```diff
@@ -197,28 +182,21 @@ def bqm_structured(f):
...
-            # todo: better error message
-            raise BinaryQuadraticModelStructureError("given bqm does not match the sampler's structure")
-        if not all(u in adjacency[v] for u, v in bqm.quadratic):
-            # todo: better error message
-            raise BinaryQuadraticModelStructureError("given bqm does not match the sampler's structure")
-
+    def structured_sample(sampler, bqm, **kwargs):
+        adjacency = sampler.adjacency
+
+        if not adjacency.keys() >= bqm.variables:
```
Comment (hhtong) [inline dimod/decorators.py:189]: Would `adjacency.keys() < bqm.variables` be better for readability?

File: dimod/decorators.py [hunk 2]
```diff
@@ -197,28 +182,21 @@ def bqm_structured(f):
...
+            raise BinaryQuadraticModelStructureError(
+                f"given bqm constains a variable, {(adjacency.keys() - bqm.variables).pop()!r}, "
+                "not supported by the structured solver")
+        if any(u not in adjacency[v] for u, v, _ in bqm.iter_quadratic()):
+            u, v = next((u, v) for u, v, _ in bqm.iter_quadratic() if u not in adjacency[v])
+            raise BinaryQuadraticModelStructureError(
+                f"given bqm constains an interaction, {(u, v)!r}, "
```
Comment (hhtong) [inline dimod/decorators.py:196]:
```suggestion
                f"given bqm contains an interaction, {(u, v)!r}, "
```

**Review by hhtong — APPROVED** [review]
LGTM

File: dimod/decorators.py (hunk 1 above)
Comment (arcondello) [reply dimod/decorators.py:189]: In this case they have a different meaning. The current syntax says "not superset", yours says "is subset".

**Pull request merged**